tools/autograd/gen_autograd_functions.py

```diff
@@ -583,14 +584,19 @@ def save_var(var: SavedAttribute, is_output: bool) -> None:
             )
             should_append_raw_getsetdef = True
             visit_name = f"{name}_"
-        elif type == BaseCType(tensorListT) or type == BaseCType(iTensorListRefT):
+        elif (
+            type == BaseCType(tensorListT)
+            or type == BaseCType(iTensorListRefT)
+            or type == VectorCType(BaseCType(tensorT))
```

Comment:
At glance this sounded a bit surprising that `Tensor[]` as a return type seems to be treated as `vector<Tensor>` unlike `Tensor[]` being `TensorList` or something equivalent as an argument.

Reply:
as a context, https://github.com/pytorch/pytorch/blob/a4cffaae675130233fe8a92cf02baad291ec39f9/tools/autograd/derivatives.yaml#L3044-L3046 seems to require this path to allow `vector<Tensor>`

Reply:
Wait is this specifically for pow? If so, its probably worth a comment.
Maybe worth a comment anyway about arguments and inputs translating into different cpp dtypes.

Reply:
4fcb0c927ba39d78f974139702b540e3da6d35a5 adding an assert and a comment